smarts/core/bubble_manager.py

```diff
@@ -315,8 +315,9 @@ def _airlock_social_vehicle_with_social_agent(
         mission_planner = MissionPlanner(
             sim.scenario.waypoints, sim.scenario.road_network
         )
+        is_boid = isinstance(social_agent_actor, BoidAgentActor)
```

Comment:
I think it might be best to leave the check for being a boid to the agent manager.

Reply:
That is the case, but routing through the vehicle index which tracks boid state. Here we're working off of SStudio types, which `AgentManager` knows nothing about.

Reply:
Ok, that seems reasonable then.